### PR TITLE
Adds an agent to reconcile the classic link security groups.

### PR DESCRIPTION
**Pull request by cfieber**
During a deploy the classiclink and app specific groups can be attached.
This agent looks for instances that do not conform and adds the security groups.

@spinnaker/netflix-reviewers PTAL

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/agent/ReconcileClassicLinkSecurityGroupsAgent.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+      .forEach(i -> {
+        List<String> existingClassicLinkGroups = i.getGroups().stream()
+          .map(GroupIdentifier::getGroupId)
+          .collect(Collectors.toList());
+
+        int maxNewGroups = deployDefaults.getMaxClassicLinkSecurityGroups() - existingClassicLinkGroups.size();
+        i.getTags()
+          .stream()
+          .filter(t -> AUTOSCALING_TAG.equals(t.getKey()))
+          .map(Tag::getValue)
```
Comment (ajordens) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/agent/ReconcileClassicLinkSecurityGroupsAgent.java]: Might be a little difficult to follow what's going on here ... maybe a quick example comment for each map:

```
.map(Tag::getValue)  // clouddriver-main-v001
...
.map(this::getSecurityGroupNames) // clouddriver
```

**Comment by ajordens** [comment]
Feels like a few inline comments around the agent would help those without intimate knowledge of what's going on.

**Comment by cfieber** [comment]
broke up stream processing into more logical chunks and added a few more test cases

**Comment by cfieber** [comment]
@ajordens let me know if this looks more reasonable

**Comment by ajordens** [comment]
LGTM

**Pull request merged**